[xiao wei] iP

xxiaoweii opened the pull request:
# Chatty 🦜 
> "The art of conversation is the art of hearing as well as of being heard." - William Hazlitt

### 🚀 Meet Chatty: Your PhD-Powered Chat Companion! 🎓 
Unleash the power of Chatty, the chatbot with a PhD in chatology **_NOW_**! 

Chatty is: 
💬 text-based:
Chatty communicates seamlessly through text, making interactions a breeze. Conversations have never been so ~~ smOOth ~~

📚 easy to learn:
Chatty is designed with simplicity in mind. Learning to use it is as easy as **1-2-3**!
_No tech wizardry_ 🧙 _required!_

💼 your **_PERSONAL_** task manager:
- [x] Manage todos 📝 
- [x] Manage deadlines 📆 
- [x] Manage events 📅 
Just need to _DOWNLOAD_ ➡️ _DOUBLE CLICK_ ➡️ and **VOILA** 🥳! Chatty becomes your trusty task manager 👨‍💼
Stay organized _effortlessly_

🤑 **_FREE_**!
That's right, Chatty's talents won't cost you a dime. 
A PhD-level chat companion for absolutely **NOTHING** 💰 

🌐 wait **NO MORE** Get started **_NOW_** 🌐 
Experience the future of chatbot 🔈 
Get Chatty [**HERE**](https://github.com/nus-cs2103-AY2324S1/ip/pull/375) and let the chat 🔮 magic 🔮 begin!

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

xxiaoweii committed:
level 0 - rename, greet, exit

xxiaoweii committed:
level 0 - rename, greet, exit

xxiaoweii committed:
level 1

xxiaoweii committed:
level 1

xxiaoweii committed:
level 1 -

xxiaoweii committed:
level 1 -

xxiaoweii committed:
level 1 - echo

xxiaoweii committed:
level 1 - echo

xxiaoweii committed:
level 1 - echo

xxiaoweii committed:
level 1 - echo

xxiaoweii committed:
level 2 - Add, List

xxiaoweii committed:
level 2 - Add, List

xxiaoweii committed:
level 3 - Mark As Done

xxiaoweii committed:
level 3 - Mark As Done

xxiaoweii committed:
level 4: ToDos, Events, Deadlines

xxiaoweii committed:
level 4: ToDos, Events, Deadlines

xxiaoweii committed:
A-TestUiTesting

xxiaoweii committed:
A-TestUiTesting

xxiaoweii committed:
Level 5 - Handle Errors

xxiaoweii committed:
Level 5 - Handle Errors

xxiaoweii committed:
Level 6 - Delete

xxiaoweii committed:
Level 6 - Delete

xxiaoweii committed:
Level 6 - updated the code slightly

xxiaoweii committed:
Level 6 - updated the code slightly

xxiaoweii committed:
Level 7 - Save

xxiaoweii committed:
Level 7 - Save

xxiaoweii committed:
Level 8 - Dates and Times

xxiaoweii committed:
Level 8 - Dates and Times

xxiaoweii committed:
Level 8 - Dates and Times

xxiaoweii committed:
Level 8 - Dates and Times

xxiaoweii committed:
Level 8 - Dates and Times

xxiaoweii committed:
Level 8 - Dates and Times

xxiaoweii committed:
A-Packages

xxiaoweii committed:
A-Packages

xxiaoweii committed:
A-Packages

xxiaoweii committed:
A-Packages

xxiaoweii committed:
Merge remote-tracking branch 'origin/master' into add-gradle-support

xxiaoweii committed:
Merge branch 'add-gradle-support'

xxiaoweii committed:
JUnit test

xxiaoweii committed:
JUnit test

xxiaoweii committed:
A-Jar

xxiaoweii committed:
A-JavaDoc

xxiaoweii committed:
A-CodingStandard

xxiaoweii committed:
Level 9 - Find

xxiaoweii committed:
Level 9 - Find

xxiaoweii committed:
A-CheckStyle

xxiaoweii committed:
A-CheckStyle

xxiaoweii committed:
A-CheckStyle

Cleon2 commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,211 @@
+ * Command is responsible for handling the different command
+ * enter by the user.
+ */
+public class Command {
```
Command should be an abstract class with only two methods: isExit() and execute(). Should create classes like FindCommand or addComand that extend Command

Cleon2 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,116 @@
+    /**
+     * Run the chatbot depending on what command the user entered
+     */
+    public void startChatting() {
```
The parsing logic should be within the parser class

Cleon2 commented on src/main/java/duke/Messages.java:
```diff
@@ -0,0 +1,48 @@
+/**
+ * Message class is responsible for printing the relevant messages
+ */
+public class Messages {
```
Print of messages should be handled by the UI class

Cleon2 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,87 @@
+     *
+     * @throws IOException if the storage file cannot be found
+     */
+    public void loadTask() throws IOException, DetailsUnknownException {
```
Should check for the case where the directory is absent

Cleon2 reviewed (commented):
Overall good attempt but I think more effort could be put into abstracting the classes properly and aligning more closely with the given requirements

Saezenn commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,111 @@
+     *
+     * @param i the index of the task to be mark as done
+     */
+    public void markAsDone(int i) {
```
can consider renaming the markAsDone and markAsUndone methods to something like markTaskAsDone and markTaskAsUndone to make their purpose more explicit.

Saezenn commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,111 @@
+     *
+     * @param index the index of the task to be removed
+     */
+    public void deleteTask(int index) {
```
can consider adding error handling for cases where users try to delete or mark tasks that do not exist (e.g., an invalid task index).

Saezenn commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,211 @@
+     */
+    public void handleDelete(String uI, String keyword) throws IncompleteMessageException, InvalidTaskNumberException {
+        if (uI.length() == keyword.length()) {
+            throw new IncompleteMessageException("Delete");
```
can consider giving clearer error messages

Saezenn commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,51 @@
+     * Constructor for Ui class
+     * Create a Scanner object
+     */
+    public Ui() {
```
Since you create a Scanner object in the constructor, consider implementing a method to close the scanner when it's no longer needed to release resources explicitly.

Saezenn commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,51 @@
+     * @return the list that contains the keyword
+     */
+    public String find(String keyword) {
+        String result = " ";
```
can consider using a StringBuilder for constructing the result string. This can be more efficient than concatenating strings in a loop, especially if there are many matching tasks.

Saezenn reviewed (commented):
LGTM! No major problems. Good documentation as well.

xxiaoweii committed:
Level 10 - GUI

xxiaoweii committed:
Added JavaDoc

xxiaoweii committed:
Update build.gradle

xxiaoweii committed:
Added assertions

xxiaoweii committed:
Update build.gradle

xxiaoweii committed:
Update build.gradle

xxiaoweii committed:
Merge branch 'branch-A-Assertions' of https://github.com/xxiaoweii/ip into branch-A-Assertions

xxiaoweii committed:
Merge pull request #2 from xxiaoweii/branch-A-Assertions

Branch-A-Assertions

xxiaoweii committed:
Improve code quality

xxiaoweii committed:
Merge pull request #3 from xxiaoweii/branch-A-CodeQuality

Improve code quality

xxiaoweii committed:
Improve code quality

xxiaoweii committed:
Merge remote-tracking branch 'origin/master'

xxiaoweii committed:
Modify code to follow coding standards

Small modifications aare made to several classes so the code comply with the given coding standard.

These changes are to enhance code quality, ensuring that the codebase remains consistent and easier for all to understand.

xxiaoweii committed:
Merge pull request #4 from xxiaoweii/branch-A-CodingStandard

Modify code to follow coding standards

xxiaoweii committed:
Add CI

Introduced Continuous Integration by adding gradle.yml file. The file will configure GitHub Actions to automate, customize and execute the workflows in GitHub repository.

Implementing CI helps to maintain code quality and reliability.

xxiaoweii committed:
Add custom alias extension

The syntax of this latest version is friendlier whereby it allows user to set their own command alias, allowing the users to personalize the chatbot's commands to match their preferences.

This enhancement provide a more user-centric experience, improving users' ability to interact with the chatbot.

xxiaoweii committed:
Enhance GUI

Several enhancements are made to the classes in the gui package to improve the GUI.

These changes aim to improve user experience and make interactions with Chatty more pleasant.

xxiaoweii committed:
Add Ui.png

xxiaoweii committed:
Enhance Friendlier-Syntax extension

The previous version of friendlier-syntax extension was updated for the chatbot to remember the alias setting that the user has set so that users do not have to re-set the alias everytime they start the chatbot.

This enhancement make it more convenient for user to use the chatbot, improving the user experience.

xxiaoweii committed:
Enhance Friendlier-Syntax extension

The previous version of friendlier-syntax extension was updated for the chatbot to remember the alias setting that the user has set so that users do not have to re-set the alias everytime they start the chatbot.

This enhancement make it more convenient for user to use the chatbot, improving the user experience.

xxiaoweii committed:
Update Ui.png

xxiaoweii committed:
Add UserGuide
